src/oneD/Boundary1D.cpp

```diff
@@ -127,23 +132,34 @@ void Inlet1D::show(const double* x)
 
 void Inlet1D::setMoleFractions(const string& xin)
 {
-    m_xstr = xin;
-    if (m_flow) {
-        m_flow->phase().setMoleFractionsByName(xin);
-        m_flow->phase().getMassFractions(m_yin.data());
+    if (m_solution) {
+        auto thermo = m_solution->thermo();
+        thermo->setMoleFractionsByName(xin);
+        thermo->getMassFractions(m_yin.data());
         needJacUpdate();
+    } else {
+        m_xstr = xin;
     }
```

Comment:
I think I understand the direction this is going -- I imagine by Cantera 3.3 we'll be requiring the `Solution` object to initialize the `Domain1D` object, so this will always be available. But I'm a little confused at what in the Matlab interface made this necessary to tackle now and could use a little clarification.